### PR TITLE
refactor(api): Changes CalibratePipetteResult offset type

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

closes https://opentrons.atlassian.net/browse/RLIQ-205.
Changed pipetteOffset result type to a Pydantic class.

# Changelog

- changed pipetteOffset result type to InstrumentOffsetVector

# Risk assessment

low. changed type of returned value. shape stayed the same

### CODE REVIEW

File: api/src/opentrons/protocol_engine/types.py [hunk 1]
```diff
@@ -214,6 +214,14 @@ class LabwareOffsetVector(BaseModel):
     z: float
 
 
+class InstrumentOffsetVector(BaseModel):
+    """Instrument Offset from home position to robot deck."""
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/types.py:214]: Is this true? I would have thought the offset would be from the instrument's nominal position to its  measured position.

File: api/src/opentrons/protocol_engine/types.py [hunk 2]
```diff
@@ -214,6 +214,14 @@ class LabwareOffsetVector(BaseModel):
...
+
+    x: float
+    y: float
+    z: float
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/types.py:218]: I would keep this exactly as you've written it for this PR, but here's a question for the group:

Protocol Engine and the HTTP API have several vector types now that are basically the same thing. `InstrumentOffsetVector`, `LabwareOffsetVector`, `ModuleCalibrationPoint`, `DeckPoint`, `WellOffset`...

Should we just have a single `class Vec3(BaseModel)` and reuse it, instead?

**Review by SyntaxColoring — APPROVED** [review]
Looks good to me if tests and CI pass. Thanks for fixing this so quickly!

File: api/src/opentrons/protocol_engine/types.py (hunk 2 above)
Comment (Laura-Danielle) [reply api/src/opentrons/protocol_engine/types.py:218]: Re-use it as in inherit from the base class?

File: api/src/opentrons/protocol_engine/types.py (hunk 1 above)
Comment (Laura-Danielle) [reply api/src/opentrons/protocol_engine/types.py:214]: The nominal position is supposed to represent where we think the instrument is with respect to the home position.

**Review by Laura-Danielle — APPROVED** [review]
Looks good to me.

File: api/src/opentrons/protocol_engine/types.py (hunk 2 above)
Comment (TamarZanzouri) [reply api/src/opentrons/protocol_engine/types.py:218]: That sounds good to me @SyntaxColoring ! I can change it for this PR or merge this one.

**Pull request merged**